### DISCUSSION TITLE
Standalone meta language server run

### DISCUSSION THREAD

**Discussion by albilu** [scalameta/metals#3685] [Q&A]
Hello,

Is it possible to start this server in standalone mode. I mean without Vscode, in  a CLI for exemple so that any other client can interact with the server.

If yes please tell how. I tried java -jar sbt-launch.jar but it doesn't seems to be that way.

Thanks you

**Comment by tgodzik** [comment]
It's not possible unfortunately since the server relies heavily on the LSP spec and currently to use it in CLI you would need to write all the jsons manually.

**Reply by albilu** [reply]
I am planning to use a proper LSP client based on lsp4j so no need to write the jsons. 
So there is no way other Editors except the one supported here https://scalameta.org/metals/docs/ can leverage this server? so sad

thanks you

**Reply by tgodzik** [reply]
The way it's usually done is that the editor starts the server and communicates via standard input/output. I think that's the default way for many language servers and any LSP aware client should be able to support it.

My comment was only about having it run without any client.

**Reply by albilu** [reply]
Exactly. that's what i am talking about. So i am looking for the way to start the server. is there any binary to launch via java or scala? 
Basically My editor based on java can start servers like this: 
```
Process p = new ProcessBuilder("python", "-m", "pylsp")
                    .redirectError(ProcessBuilder.Redirect.INHERIT)
                    .start();
```
This is an example for python based lsp (python -m pylsp) . So i am looking for similar command to start Metals if its possible

**Reply by tgodzik** [reply]
There is a bit more info here https://scalameta.org/metals/docs/integrations/new-editor
